hack/test-go.sh

```diff
@@ -77,22 +77,45 @@ fi
 
 KUBE_TIMEOUT=${KUBE_TIMEOUT:--timeout 60s}
 
-if [ "${1-}" != "" ]; then
-  if [[ "${1}" == *"/..." ]]; then
+# Declare and set arrays to null string. We need to always keep the first
+# element to be the null string, otherwise the variable is considered unset.
+# When using, slice it to ignore the first element: ${varname[@]:1}.
+declare -a test_flags= test_packages=
```

Comment:
This is definitely thorny. I think your implementation is the best we can do unless we change the API for `test-go.sh`.